### PR TITLE
fix: when enable or disable existing SSL, an error occurred

### LINKED ISSUE #1034: bug: when enable or disable existing ssl certificate,  will be error
# Bug report

## Describe the bug

when enable or disable existing ssl certificate,  will be error
```
$ curl http://127.0.0.1:9000/apisix/admin/ssl/332467865074533428 -d ‘{“status”:1}’ -X PATCH
2020-12-12 11:52:22.052345 I | http: panic serving 127.0.0.1:53773: runtime error: invalid memory address or nil pointer dereference
goroutine 70 [running]:
net/http.(*conn).serve.func1(0xc0003005a0)
	/usr/local/go/src/net/http/server.go:1801 +0x147
panic(0x19a0780, 0x212d5b0)
	/usr/local/go/src/runtime/panic.go:975 +0x3e9
github.com/apisix/manager-api/filter.Authentication.func1(0xc00035e690)
	... ...
curl: (52) Empty reply from server
```

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
fix #1034

___
### Bugfix
- Description
The SSL `GET` api does not return the SSL key, so set `ssl.key` to nil, which will be saved in the cache. 
At this time, use the patch method to update the SSL, because the lack of the key, it fails the schema verification, and return an error.
The `ErrorWrapper` use it directly without judging the type of error, which eventually leads to panic.

- How to fix?

Set the key to nil after clone `ssl` to avoid affecting the data in the cache.
After judging the error type, do the corresponding treatment before use.

### CODE REVIEW

File: api/internal/utils/consts/api_error.go [hunk 1]
```diff
@@ -27,7 +29,21 @@ func ErrorWrapper(handle WrapperHandle) gin.HandlerFunc {
 	return func(c *gin.Context) {
 		data, err := handle(c)
 		if err != nil {
-			apiError := err.(*ApiError)
+			apiError, ok := err.(*ApiError)
+			if !ok {
+				errMsg := err.Error()
+				if strings.Contains(errMsg, "required") ||
```
Comment (tokers) [inline api/internal/utils/consts/api_error.go:35]: Why not encapsulating these logics into a separate method in `ApiError`, like `ApiError.IsInvalidParam`?

File: api/internal/utils/utils.go [hunk 2]
```diff
@@ -97,6 +98,16 @@ func InterfaceToString(val interface{}) string {
 	return str
 }
 
+func ObjectClone(origin, copy interface{}) error {
```
Comment (tokers) [inline api/internal/utils/utils.go:101]: We'd better to add some comments for this function, such as the object should not have float number, the `json.Marshal` and `json.Unmarshal` will cause the precision loss.

File: api/internal/utils/consts/api_error.go (hunk 1 above)
Comment (starsz) [reply api/internal/utils/consts/api_error.go:35]: I had seen the same code in `hadler.go`.
https://github.com/apache/apisix-dashboard/blob/cb1377cf9859392ee81ee60b3d4e623b1522be1e/api/internal/handler/handler.go#L55-L61

So I think we'd better discuss about the response error.

File: api/internal/handler/ssl/ssl.go [hunk 3]
```diff
@@ -86,7 +86,11 @@ func (h *Handler) Get(c droplet.Context) (interface{}, error) {
 	}
 
 	//format respond
-	ssl := ret.(*entity.SSL)
+	ssl := &entity.SSL{}
```
[the remark this answers is not in the document]
Comment (spacewander) [reply api/internal/handler/ssl/ssl.go:89]: I remember the literal struct initialization is encouraged...
https://github.com/uber-go/guide/blob/master/style.md#initializing-struct-references

**Comment by membphis** [comment]
@nic-chen need E2E test cases to confirm this bug ^_^

File: api/internal/handler/ssl/ssl.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (membphis) [reply api/internal/handler/ssl/ssl.go:89]: @nic-chen ping

[the remark this answers is not in the document]
Comment (nic-chen) [reply api/internal/handler/ssl/ssl.go:89]: @tokers we do need an empty entity.SSL object here. what's the recommend way to create an empty object?

File: api/internal/utils/utils.go (hunk 2 above)
Comment (nic-chen) [reply api/internal/utils/utils.go:102]: added.

**Comment by nic-chen** [comment]
> @nic-chen need E2E test cases to confirm this bug ^_^

@membphis 
we already have E2E test cases  for it:
https://github.com/nic-chen/incubator-apisix-dashboard/blob/fix-ssl-status-bug/api/test/e2e/ssl_test.go#L77-L132

just need to add a test case to trigger the bug:
https://github.com/apache/apisix-dashboard/pull/1064/files#diff-c2b7edfd95881ffeb7a02870b56d8c228b7b1f0f83d78be56f4462091bf76dc7R104-R111

File: api/internal/handler/ssl/ssl.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShiningRush) [reply api/internal/handler/ssl/ssl.go:89]: I think it is fine to use `var := &object{}`,and  it is used in many open-source projects.

File: api/internal/utils/consts/api_error.go (hunk 1 above)
Comment (nic-chen) [reply api/internal/utils/consts/api_error.go:35]: @starsz  sure, we need a PR to optimize it.

File: api/internal/handler/ssl/ssl.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tokers) [reply api/internal/handler/ssl/ssl.go:89]: > I remember the literal struct initialization is encouraged...
https://github.com/uber-go/guide/blob/master/style.md#initializing-struct-references

my bad.

**Review by tokers — APPROVED** [review]

**Review by gxthrj — APPROVED** [review]

**Pull request merged**